### PR TITLE
Add device zyhps01_human_presence_sensor

### PR DESCRIPTION
**Pull request by EnzoD86**
Hi, I purchased this presence sensor:

https://it.aliexpress.com/item/1005007130244006.html?spm=a2g0o.order_list.order_list_main.5.1c7a3696PuBlcU&gatewayAdapt=glo2ita

I would like to add it to the repository.

The device does not have a standard instruction set; to be able to extract the DP's I had to switch from "Standard instruction set" to "DP instruction" on the Tuya IOT platform

I attach the extracted DP's.

```
{
  "result": {
    "category": "kg",
    "functions": [
      {
        "code": "presence_delay",
        "dp_id": 104,
        "type": "Integer",
        "values": "{\"unit\":\"s\",\"min\":0,\"max\":180,\"scale\":0,\"step\":1}"
      },
      {
        "code": "movesensitivity",
        "dp_id": 105,
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "breathsensitivity",
        "dp_id": 107,
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "movedistance_max",
        "dp_id": 109,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      },
      {
        "code": "movedistance_min",
        "dp_id": 110,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      },
      {
        "code": "breathdistance_max",
        "dp_id": 111,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      },
      {
        "code": "breathdistance_min",
        "dp_id": 112,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      }
    ],
    "lang_config": {},
    "status": [
      {
        "code": "presence_state",
        "dp_id": 101,
        "type": "Enum",
        "values": "{\"range\":[\"presence\",\"none\"]}"
      },
      {
        "code": "illuminance_value",
        "dp_id": 103,
        "type": "Integer",
        "values": "{\"unit\":\"lux\",\"min\":0,\"max\":3000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "presence_delay",
        "dp_id": 104,
        "type": "Integer",
        "values": "{\"unit\":\"s\",\"min\":0,\"max\":180,\"scale\":0,\"step\":1}"
      },
      {
        "code": "movesensitivity",
        "dp_id": 105,
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "breathsensitivity",
        "dp_id": 107,
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "movedistance_max",
        "dp_id": 109,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      },
      {
        "code": "movedistance_min",
        "dp_id": 110,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      },
      {
        "code": "breathdistance_max",
        "dp_id": 111,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      },
      {
        "code": "breathdistance_min",
        "dp_id": 112,
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":10}"
      }
    ]
  },
  "success": true,
  "t": 1723165756272,
  "tid": "ff66543855eb11ef8d1002b5f94b0719"
}
```

The yaml file I created and inserted into the pull request, I tested it and it works perfectly.
I only have a doubt about the class of the sensor inserted in the primary entity.
I used a "motion" class even though this is a presence sensor.
After doing several tests, the motion class still seemed the most appropriate to me.
Because using an "occupancy" class the translation was horrible... using the "presence" class interprets it as home or not home (when instead the sensor refers to a room).

Do you agree?
Thank you

### CODE REVIEW

**Comment by make-all** [comment]
For a room, the class should be occupancy.  If the translation into your language is not good, you should report it upstream to Home Assistant.

**Comment by EnzoD86** [comment]
Yes, I agree with you. Do you prefer me to make the change or you to do it?

File: custom_components/tuya_local/devices/zyhps01_human_presence_sensor.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+name: Human presence sensor
+products:
+  - id: wzp9yjsobanavsut
+    name: Human Presence Sensor
```
Comment (make-all) [inline custom_components/tuya_local/devices/zyhps01_human_presence_sensor.yaml:4]: This name should be the manufacturer (if known) and model (ZY-HPS-01?) of the product.

**Review by make-all — COMMENTED** [review]
Since there are adjustments for both breath and motion sensitivity and distance, is there another sensor for motion, or a third state alongside presence and none for indicating motion?

**Comment by EnzoD86** [comment]
as you can see from the json I attached, there is only one exposed sensor and it only has two states.

Actually in the json the fields are called breath, but in the smart life settings it says "micro-motion".






regarding the model, I confirm that this is the ZY-HPS01-5.8G but I don't see the product name on the box

**Pull request merged**